internals/overlord/overlord.go

```diff
@@ -321,6 +328,25 @@ func (o *Overlord) ensureTimerSetup() {
 	o.pruneTicker = time.NewTicker(pruneInterval)
 }
 
+func (o *Overlord) ensureTimerStopClearChannel() {
+	o.ensureLock.Lock()
+	defer o.ensureLock.Unlock()
```

Comment:
Do we actually need to acquire this lock here?

Reply:
I believe so, because now ```state.EnsureBefore()``` can be called much earlier than when this is called.

Reply:
Function removed. Resolving.